src/distributors/local-installer.ts

```diff
@@ -0,0 +1,60 @@
+import * as tc from '@actions/tool-cache';
+
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+import { macOSJavaContentDir, setupFromJdkFile } from '../util';
+
+import { JavaBase } from './base-installer';
+import {
+  JavaInstallerOptions,
+  JavaDownloadRelease,
+  JavaInstallerResults
+} from './base-models';
+
+export class LocalDistributor extends JavaBase {
+  private jdkFile: string;
+  constructor(installerOptions: JavaInstallerOptions) {
+    super('local_distributor', installerOptions);
+    this.jdkFile = installerOptions.jdkFile!;
+  }
+
+  protected async findPackageForDownload(
+    version: semver.Range
+  ): Promise<JavaDownloadRelease> {
+    throw new Error('Should not be implemented');
+  }
+
+  protected async downloadTool(
+    javaRelease: JavaDownloadRelease
+  ): Promise<JavaInstallerResults> {
+    throw new Error('Should not be implemented');
+  }
```

Comment:
Let's move these functions to bottom of class

Reply:
Done. Thank you.